[27기 김태영] Add: 슬라이드 넘버 버튼 기능 구현 중

Moro-yong opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 가로 스크롤 기능
- 스크롤 버튼 기능
- 마우스 오버 이미지 사이즈 크기 조절 기능 
- 새로운 글 업로드시 알림 기능
- 가로 무한스크롤 기능
- 태그에 따른 내용 분류 기능

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
- html, css 활용하여 레이아웃 구성
- top 슬라이드 이미지 hover 기능 추가 (hover 시 이미지 스케일 증가)
- top 슬라이드 좌우 버튼 클릭 기능 추가
- ( window.scrollY )을 활용하여 최상단으로 이동 버튼 기능 추가
- 

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 여러가지 경로에서 모르는 기능을 찾기 위해 다양한 검색어로 검색하고 함수형과 클래스형의 차이를 비교하면서 함수형에 맞도록 수정하는 과정이 매우 힘들었습니다. 
- window.scrollY / Window.pageYOffset의 기능과 사용법
- 함수를 더 짧게 작성하는 여러가지 방법(조건부 랜더링 등)
- 프롭스로 내려준 함수를 활용하여 자식 컴포넌트에서 데이터 값을 변경할 경우 부모 컴포넌트의 데이터 값도 변경 되는 점

<br />

## :: 기타 질문 및 특이 사항
- 너무 어렵네요 슬라이드 밑에 번호를 누르면 해당 이미지로 이동하도록 하고 싶은데 모르겠습니다 ㅠ

Moro-yong committed:
Add: branch site - Main page 추가
 - 레이아웃 구조 추가
 - scss 추가

Moro-yong committed:
Add: 상단 슬라이드 이미지 hover 추가

lang92 commented on public/data/feedData.json:
```diff
@@ -0,0 +1,26 @@
+[
+  {
+    "src": "/taeyoung/images/userImg1.jpg",
+    "id": "feed_img1",
+    "userName": "kkasmne1453",
+    "likePeople": "431"
+  },
+  {
+    "src": "/taeyoung/images/erick-butler-8lZzSaNPuPc-unsplash.jpg",
+    "id": "feed_img2",
+    "userName": "han",
+    "likePeople": "41"
+  },
+  {
+    "src": "/taeyoung/images/mathias-p-r-reding-5bwYLeO4ViA-unsplash.jpg",
+    "id": "feed_img3",
+    "userName": "park_askwk23",
+    "likePeople": "1671"
+  },
+  {
+    "src": "/taeyoung/images/feedImg.png",
+    "id": "feed_img4",
+    "userName": "lee_oh",
+    "likePeople": "2"
+  }
+]
```
westagram에서 사용하셨던 mock Data인가요..!?

Moro-yong replied:
앗 맞습니다. 위스타그램에서 쓰던 목업데이터 변형해서 쓰다가 못 지운 파일이네요 지우겠습니다

lang92 commented on src/pages/main/Articles/Articles.js:
```diff
@@ -0,0 +1,106 @@
+
+function Articles() {
+  return (
+    <div className="articles">
+      <section className="articlesContainer">
```
div, section 태그 두 개 중 하나만 사용해도 될 것 같습니다!

Moro-yong replied:
넵 수정하겠습니다!

lang92 commented on src/pages/main/Articles/Articles.js:
```diff
@@ -0,0 +1,106 @@
+// import React, { useState, useEffect } from 'react';
+import SlideButton from '../SlideButton/SlideButton';
+
```
빈 줄 지워주세요!

lang92 commented on src/pages/main/Articles/Articles.js:
```diff
@@ -0,0 +1,106 @@
+  return (
+    <div className="articles">
+      <section className="articlesContainer">
+        <div className="articlesbox">
+          <img
+            className="articlesUserFeedImg"
+            src="/images/Main/microsoft-edge-FAaz8lkinzs-unsplash.jpg"
+            alt="NewFeed"
+          />
+          <span className="articlesUserFeedTitle">
+            스타벅스에서 톨사이즈 주문하다 까여본적 있나요?
+          </span>
+          <span className="articlesUserFeedSub">
+            맥시멀 리스트 남편이 또 뭔가를 샀다. 사촌 집에서 본 크롬캐스트를
+            사서 HBO로 요즘 핫하다는 듄을 봤다. 아주 먼 미래의 이야기라 주인공이
+            걸어 다니면 조명이 따라다닌다. 티모시가 나오는데도 불구하고 나는
+            중간에 박차고 나왔다. 아이맥스로 봐야 재밌다는 걸 집에서 봐서 그런
+            걸까?
+          </span>
+          <span className="articlesUserFeedName">by 작성자</span>
+        </div>
```
반복되는 UI네요! 컴포넌트화해서 사용할 수 있을 것 같습니다

lang92 commented on src/pages/main/KeyWord/KeyWord.js:
```diff
@@ -0,0 +1,26 @@
+// import React, { useState, useEffect } from 'react';
+
```
여기도!

lang92 replied:
이외에 리뷰하지 않은 부분도 수정해주세요!

lang92 commented on src/pages/main/KeyWord/KeyWord.scss:
```diff
@@ -0,0 +1,35 @@
+  }
+
+  .keyWordContainer {
+    .keyWordContainerLine1,
+    .keyWordContainerLine2 {
```
동일한 style 속성을 가지고 있다면, className 분리하지 않아도 될 것 같습니다.

Moro-yong replied:
나중에 keyWordContainerLine1과 keyWordContainerLine2의 내부 레이아웃을 변경하려고 미리 바꿔둔건데 그건 그때가서 바꿔야겠네요 감사합니다!

lang92 commented on src/pages/main/Main.js:
```diff
@@ -0,0 +1,41 @@
+import KeyWord from './KeyWord/KeyWord';
+import Writers from './Writers/Writers';
+import Articles from './Articles/Articles';
+
```
여기도~~

lang92 commented on src/pages/main/Main.js:
```diff
@@ -0,0 +1,41 @@
+        <span className="topButtonText">
+          Top
+          <img
+            className="TopButtonImg"
```
camelCase로 작성해주세요!

lang92 commented on src/pages/main/Main.js:
```diff
@@ -0,0 +1,41 @@
+
+  return (
+    <div className="main">
+      <Slide className="componuntSlideImage" userData={userData} />
+      <KeyWord className="componuntKeyWord" />
+      <Writers className="componuntWriters" />
+      <Articles className="componuntArticles" />
```
컴포넌트 단위로 잘 나눠주셨네요!
하지만 component에는 className을 부여하실 수 없습니다.
해당하는 컴포넌트의 props로 className이라는 key와 해당 value가 전달되었을 것 같네요!

Moro-yong replied:
props를 console.log로 확인하면 부여한 className가 뜨지 않네요
감사합니다 새로운 지식을 습득했습니다!

lang92 commented on src/pages/main/Slide/Slide.js:
```diff
@@ -0,0 +1,17 @@
+  const { userData } = props;
+
+  return (
+    <div className="Slide">
```
camelCase로 작성해주세요! 다른 부분도 찾아서 수정해주세요

lang92 commented on src/styles/common.scss:
```diff
@@ -1,10 +1,11 @@
-  font-family: "Noto Sans KR", sans-serif;
+  font-family: 'Noto Sans KR', sans-serif;
   font-size: 14px;
+  font-weight: normal;
```
팀원들과 협의된 내용인가요??

lang92 commented on src/pages/main/KeyWord/KeyWord.js:
```diff
@@ -0,0 +1,26 @@
+
+function KeyWord() {
+  return (
+    <span className="keyWord">
+      <span className="keyWordTitle">BRANCH KEYWORD</span>
+      <span className="keyWordSubText">키워드로 분류된 다양한 글 모음</span>
+      <section className="keyWordContainer">
+        <div className="keyWordContainerLine1">
+          <div className="keyWordContents">js</div>
+          <div className="keyWordContents">js</div>
+          <div className="keyWordContents">js</div>
+        </div>
+        <div className="keyWordContainerLine2">
+          <div className="keyWordContents">js</div>
+          <div className="keyWordContents">js</div>
+          <div className="keyWordContents">js</div>
+        </div>
+      </section>
+    </span>
```
className을 명확히 써주시는 건 좋지만,
keyWord라는 className으로 nesting했기 때문에 모든 className마다 keyWord라고 써주지 않아도 될 것 같습니다!

lang92 commented on src/pages/main/Writers/Writers.scss:
```diff
@@ -0,0 +1,83 @@
+      }
+
+      .writersPeopleName {
+        padding: 30px 0 20px 0;
```
다음과 같이 쓸 수 있습니다!
```suggestion
        padding: 30px 0 20px;
```

lang92 reviewed (commented):
고생하셨습니다 태영님!

기능 구현 이전에 리뷰 반영 우선적으로 해주세요 🙌

Moro-yong committed:
Add: top slider 기능 추가, Top button 기능 추가
modify: top slider 버튼 위치 수정, 목 데이터 갯수 지정하여 불러오도록 수정, 기타 css 수정

ghost commented:
Sider has detected **1** warning on analyzing the commit 7f332d5.

- ⚠️ The number of issues (46) exceeded the limit of comments. Please visit [Sider](https://sider.review/gh/repos/432938814/pulls/2) to review the issues.

If the errors persist even after [retrying](https://sider.review/gh/repos/432938814/pulls/2#tools), the following actions may resolve them:
- Update tool dependencies
- Configure failed tools
- [Configure `sider.yml`](https://sider.review/gh/repos/432938814/pulls/2#config)
- [Turn off unused tools](https://sider.review/gh/repos/432938814/settings)

If you still have problems, feel free to ask us via chat. 💬

---

You can [turn off such notifications](https://sider.review/gh/repos/432938814/settings#notifications) if unnecessary.

Moro-yong committed:
Delete: 불필요한 js파일 삭제

Moro-yong committed:
Add: 키워드 목데이터 추가, 마우스 오버 관련 css 기능 추가

Modify: 키워드 영역 반복되는 코드 map으로 수정

Moro-yong committed:
Modify: 클래스명/슬라이드 로직 변경

Moro-yong committed:
Modify:
1. 목 데이터 수정
	- 실제 데이터와 비슷한 형테로 목 데이터 생성
2. 이미지 슬라이드 컴포넌트 생성
	- 슬라이드 이미지만 분리하여 컴포넌트 생성

Moro-yong committed:
Modify: 슬라이드 로직 변경 및 슬라이드 관련 변수명 변경

Moro-yong committed:
Modify: Top이동 버튼 로직 수정 중, Writers 로직 수정

Moro-yong committed:
Delete: console.log

Moro-yong committed:
Modify: 스크롤 버튼 위치 수정

Moro-yong committed:
Delete: console.log

Moro-yong committed:
Delete: console.log

Moro-yong committed:
Modify: top버튼 로직 수정, 태그 로직 수정

Moro-yong committed:
중간저장

Moro-yong committed:
Delete: 데이터 연결 전 불필요한 로직 주석처리 - 추후 삭제예정

Moro-yong committed:
Modify: 로직 수정

Moro-yong committed:
Fix: conflict 해결

Moro-yong committed:
Fix: conflict 수정
